Peers Window rename 'Peer id' to 'Peer'

jarolrod opened the pull request:
Picking up https://github.com/bitcoin-core/gui/pull/290

**Original PR Description:**
- renames the peers tab column header from `Peer Id` to `Peer` to allow resizing the column more tightly (this will be particularly useful after #256) and does the same for the peer details area.

While here, we also add Qt translator comments for the Peer Table columns.

| Master        | PR               |
| ----------- | ----------- |
| ![Screen Shot 2021-05-03 at 1 23 05 AM](https://user-images.githubusercontent.com/23396902/116843818-20a14b00-abaf-11eb-913e-ddff11cda5cd.png) | ![Screen Shot 2021-05-05 at 4 08 45 AM](https://user-images.githubusercontent.com/23396902/117112825-a2cc7380-ad57-11eb-939b-1aceb4214ad1.png) |

hebasto commented:
@jarolrod 

Did you consider how it will look on a long-running node when one-digit id peer is next to, say, four-digit id one?

jarolrod commented:
@hebasto Actually, that's a good point. As a tabular figure, perhaps it makes more sense to leave as-is. Especially, if we enforce a monospace font at some point

jarolrod commented:
Update from 4ed95c0 -> 73a91c6

Changes: 
- Keep Peer # left-aligned as this is more appropriate for a tabular figure

hebasto commented:
> * renames the peers tab column header from `Peer Id` to `Peer` to allow resizing the column more tightly

If saving of the horizontal space is the motivation, leaving just "ID" or "id" is strictly better approach for the following reasons:
- the translated "Peer" could become much longer, e.g., the both variants in Ukrainian are "Вузол" or "З'єднання"
- "ID"/"id" is shorter, and often remains untranslated

jonatack commented:
> leaving just "ID" or "id" is strictly better approach

Hm, I disagree for the reasons I already described in the parent PR.

For laypeople (not devs or database people), the right word here is probably "number", "peer number" or just "peer", as "number" is self-understood. 

"ID" means "identification" and in common use, your ID card. "Show me some ID" is what the police officer or the security guard would ask you.

"Id" is a word from Freudian psychology IIRC.

Maybe a good solution is to have native speakers of each language decide on the best word for that language, and in cases where the English-as-a-first-language and English-as-a-second-language-or-as-dev-jargon differ, provide helpful translation hints in the translator comments.

RandyMcMillan commented:
№ - numero will work..

https://en.wikipedia.org/wiki/Numero_sign

https://github.com/RandyMcMillan/gui/tree/numero-column-name

https://github.com/RandyMcMillan/gui/commit/23dbd90e32127f72cd9a6e2dfec3909083df2e37#diff-3fcbd1d49f399ac9b4656011ea03e7ef80a855482d6de33855e6326ec065321fR77

![Screen Shot 2021-05-11 at 12 28 46 AM](https://user-images.githubusercontent.com/152159/117758494-27a00d00-b1f0-11eb-9556-7a6ea2f08711.png)

jonatack commented:
> № - numero will work..
> 
> https://en.wikipedia.org/wiki/Numero_sign

Interesting, it is a standard abbreviation in French, TIL it exists in English. I like it better than ID, not sure how well known it is and what the GUI policy is regarding acronyms and abbreviations.

hebasto commented:
Agree to use any neutral replacement for "id" string, even just "#" for English locale :)
https://en.wikipedia.org/wiki/Number_sign

jonatack commented:
These column headers are words, so for the English I would suggest using "Peer" and then describe alternatives ("This column means the peer number or ID") for translators in the tr comment.

hebasto commented:
It seem this PR needs translation comments only :)

promag commented:
Honest opinion here, could be empty. Otherwise ID or # LGTM.

RandyMcMillan commented:
In this old PR #135 "ID" looked ok...

![](https://user-images.githubusercontent.com/152159/100163321-6a10a000-2e83-11eb-90d5-78d327d32fdf.png)

jarolrod commented:
Updated from 73a91c63ec72e62ef76fbc857baff14b099a1358 -> b405f19c953d9cd5aa90b66001c31cc58230fb78 ([pr311.02](https://github.com/jarolrod/gui/commits/pr311.02), [pr311.03](https://github.com/jarolrod/gui/commits/pr311.03), [diff](https://github.com/jarolrod/gui/compare/master...jarolrod:pr311.03))

Changes:
- added translator comments for the columns of the Peers Table to address this [comment](https://github.com/bitcoin-core/gui/pull/311#issuecomment-839943546). Hopefully, this translator comment is enough to guide the translation of the term and can appease all sides on the debate of `ID` vs `Peer`

hebasto commented on src/qt/peertablemodel.h:
```diff
@@ -74,7 +74,33 @@ public Q_SLOTS:
+            unique number used to identify a connection. */
+        tr("Peer"),
+        /*: Title of Peers Table column which contains the
+            IP/Onion/I2P address we used to connect to the peer. */
```
"we used to connect to the peer" -- looks like it is about outgoing connections only.

jarolrod replied:
what do you think about this rephrasing:
 `Title of the Peers Table column which contains the IP/Onion/I2P address of the connected peer`

hebasto replied:
lgtm

jonatack replied:
"column for the peer address" ?

(reviewing from my phone so apologies if a bit out of context)

hebasto commented on src/qt/peertablemodel.h:
```diff
@@ -74,7 +74,33 @@ public Q_SLOTS:
+        /*: Title of Peers Table column which contains the
+            IP/Onion/I2P address we used to connect to the peer. */
+        tr("Address"),
+        /*: Title of Peers Table column which contains a value for
+            the type of connection we have established with the peer.
```
"a value for the type" -- sounds unusual for me.

jonatack replied:
agree that "we have established" sounds like outbound

"which describes the type of peer connection" ?

hebasto commented on src/qt/peertablemodel.h:
```diff
@@ -74,7 +74,33 @@ public Q_SLOTS:
+            the type of connection we have established with the peer.
+            Type is a term denoting what the connection is for. */
+        tr("Type"),
+        /*: Title of Peers Table column which contains a value for
+            the network the peer resides on. */
```
"a value for the network" -- sounds unusual for me.

jonatack replied:
"column for the network the peer connected through" ?￼

hebasto commented on src/qt/peertablemodel.h:
```diff
@@ -74,7 +74,33 @@ public Q_SLOTS:
+            the latency of the connection established with the peer. */
+        tr("Ping"),
+        /*: Title of Peers Table column which contains a value for
+            the total amount of bytes we have sent to the peer. */
```
"bytes" -- could the exact name of the unit be avoided? (two places)

hebasto commented on src/qt/peertablemodel.h:
```diff
@@ -74,7 +74,33 @@ public Q_SLOTS:
+        tr("Received"),
+        /*: Title of Peers Table column which contains the peer's
+            User Agent string. A peer can configure this value to
+            allow itself to be identifiable. */
```
I assume there could be more reasons to use `-uacomment`, maybe avoid this particular pattern to make the comment more general?

hebasto reviewed (approved):
ACK b405f19c953d9cd5aa90b66001c31cc58230fb78, I agree on column title renaming, and as a translator I'm happy to see more translator comments.

I'm not an expert in the English Grammar, therefore, the following nits could be just ignored.

jarolrod commented:
updated from b405f19 -> 657b33e ([pr311.03](https://github.com/jarolrod/gui/tree/pr311.03), [pr311.04](https://github.com/jarolrod/gui/tree/pr311.04), [diff](https://github.com/jarolrod/gui/compare/pr311.03...jarolrod:pr311.04))

**Changes:**
Updated to address review comments on the translator comment contents, thanks @hebasto and @jonatack!

Here is a diff of how I've updated from [pr311.03](https://github.com/jarolrod/gui/tree/pr311.03):

``` diff
diff --git a/src/qt/peertablemodel.h b/src/qt/peertablemodel.h
index 268ae013e..3d195342f 100644
--- a/src/qt/peertablemodel.h
+++ b/src/qt/peertablemodel.h
@@ -79,27 +79,25 @@ private:
             unique number used to identify a connection. */
         tr("Peer"),
         /*: Title of Peers Table column which contains the
-            IP/Onion/I2P address we used to connect to the peer. */
+            IP/Onion/I2P address of the connected peer. */
         tr("Address"),
-        /*: Title of Peers Table column which contains a value for
-            the type of connection we have established with the peer.
-            Type is a term denoting what the connection is for. */
+        /*: Title of Peers Table column which describes the type of
+            peer connection. The "type" describes why the connection exists. */
         tr("Type"),
-        /*: Title of Peers Table column which contains a value for
-            the network the peer resides on. */
+        /*: Title of Peers Table column which states the network the peer
+            connected through. */
         tr("Network"),
-        /*: Title of Peers Table column which contains a value for
-            the latency of the connection established with the peer. */
+        /*: Title of Peers Table column which indicates the current latency
+            of the connection with the peer. */
         tr("Ping"),
-        /*: Title of Peers Table column which contains a value for
-            the total amount of bytes we have sent to the peer. */
+        /*: Title of Peers Table column which indicates the total amount of
+            network information we have sent to the peer. */
         tr("Sent"),
-        /*: Title of Peers Table column which contains a value for
-            the total amount of bytes we have received from the peer. */
+        /*: Title of Peers Table column which indicates the total amount of
+            network information we have received from the peer. */
         tr("Received"),
         /*: Title of Peers Table column which contains the peer's
-            User Agent string. A peer can configure this value to
-            allow itself to be identifiable. */
+            User Agent string. */
         tr("User Agent")};
     std::unique_ptr<PeerTablePriv> priv;
     QTimer *timer;
```

jonatack commented:
utACK 657b33ef2de77acd1061cdf4d1d64d0e086d75df

hebasto reviewed (approved):
re-ACK 657b33ef2de77acd1061cdf4d1d64d0e086d75df
